src/AutoRest.CSharp.V3/Output/Builders/ClientBuilder.cs

```diff
@@ -248,7 +248,12 @@ private RestClientMethod BuildMethod(Operation operation, string clientName, IRe
                             }
                             else
                             {
-                                var serialization = _serializationBuilder.Build(httpRequestWithBody.KnownMediaType, requestParameter.Schema, requestParameter.IsNullable());
+                                VirtualParameter[] flattenedParameters = Array.Empty<VirtualParameter>();
+                                if (requestParameter.Flattened == true)
+                                {
+                                    flattenedParameters = parameters.OfType<VirtualParameter>().Where(vp => vp.OriginalParameter == requestParameter).ToArray();
```

Comment:
This gets the VirtualParameters (those that have been flattened) based on the original requestParameter.